tools/publisher/src/subcommand/publish.rs

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use crate::cargo;
+use crate::fs::Fs;
+use crate::package::{discover_package_batches, PackageBatch};
+use crate::repo::discover_repository;
+use crate::{REPO_CRATE_PATH, REPO_NAME};
+use anyhow::Result;
+use dialoguer::Confirm;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::Semaphore;
+use tracing::info;
+
+const BACKOFF: Duration = Duration::from_millis(30);
+const MAX_CONCURRENCY: usize = 4;
+
+pub async fn subcommand_publish() -> Result<()> {
+    // Make sure cargo exists
+    cargo::confirm_installed_on_path().await?;
+
+    info!("Discovering crates to publish...");
+    let repo = discover_repository(REPO_NAME, REPO_CRATE_PATH)?;
+    let batches = discover_package_batches(Fs::Real, &repo.crates_root).await?;
+    info!("Crates discovered.");
```

Comment:
it'd be cool if this could print the number of runtime crates and number of sdk crates just as a sanity check helper

Reply:
I like this idea! Added.